### PR TITLE
Merge multiple cookie headers coming from HTTP/2 client into a single Cookie header for HTTP/1.1 backend

### PR DESCRIPTION
**Pull request by dmpetroff**
Also fix "strange" error codes coming from bitwise OR'ing or Exx error codes.

This PR is passing `http2_general.test_h2_headers.TestSplitCookies.test_split_cookies` test. I've also checked that calculated HTTP/1.1 header size (`h1_hdrs_sz`) is equal to actual  header length intercepted by wireshark.

PR with enabling tests: https://github.com/tempesta-tech/tempesta-test/pull/469

### CODE REVIEW

**Review by const-t — APPROVED** [review]
LGTM with little cleanup.

File: fw/http.c [hunk 1]
```diff
@@ -3624,6 +3624,43 @@ tfw_h2_req_set_loc_hdrs(TfwHttpReq *req)
...
+	static const DEFINE_TFW_STR(crlf, S_CRLF);
+	TfwStr *dup, *dup_end;
+	const TfwStr *cookie_dlm = &h_cookie;
+
+	TFW_STR_FOR_EACH_DUP(dup, hdr, dup_end) {
+		TfwStr *chunk, *chunk_end, hval = {};
+
+		if (unlikely(TFW_STR_PLAIN(dup))) {
+			return -EINVAL;
+		}
```
Comment (krizhanovsky) [inline fw/http.c:3644]:
```suggestion
		if (unlikely(TFW_STR_PLAIN(dup)))
			return -EINVAL;
```
Please read the coding style guide

File: fw/http.c [hunk 2]
```diff
@@ -3624,6 +3624,43 @@ tfw_h2_req_set_loc_hdrs(TfwHttpReq *req)
...
+static int
+write_merged_cookie_headers(TfwStr *hdr, TfwMsgIter *it)
+{
+	int r = 0;
+	static const DEFINE_TFW_STR(h_cookie, "cookie" S_DLM);
+	static const DEFINE_TFW_STR(val_dlm, "; ");
+	static const DEFINE_TFW_STR(crlf, S_CRLF);
+	TfwStr *dup, *dup_end;
+	const TfwStr *cookie_dlm = &h_cookie;
+
```
Comment (krizhanovsky) [inline fw/http.c:3638]: Let's write `cookie` before the loop and we won't need change `cookie_dlm` value as well as the whole variable

File: fw/http.c [hunk 3]
```diff
@@ -3839,6 +3882,12 @@ tfw_h2_adjust_req(TfwHttpReq *req)
 				continue;
 			}
 			break;
+		case TFW_HTTP_HDR_COOKIE:
+			if (TFW_STR_DUP(field)) {
+				r |= write_merged_cookie_headers(&ht->tbl[TFW_HTTP_HDR_COOKIE], &it);
+				continue;
+			}
+			break;
```
Comment (krizhanovsky) [inline fw/http.c:3890]:
```suggestion
			if (!TFW_STR_DUP(field))
			        break;
			r |= write_merged_cookie_headers(&ht->tbl[TFW_HTTP_HDR_COOKIE], &it);
			continue;
```

Comment (krizhanovsky) [reply fw/http.c:3890]: The good point about line width limit is that it also requires to keep the code with smaller nesting (e.g. see also Code Complete book).

**Review by krizhanovsky — APPROVED** [review]
LGTM: I don't see any severe issues with the pull request - just please fix the coding style and let's merge

File: fw/http.c (hunk 2 above)
Comment (dmpetroff) [reply fw/http.c:3638]: That would just make code a bit worse as you'll have to track if separator is needed to be written or not.

**Comment by dmpetroff** [comment]
Closes #1736

**Pull request merged**